src/Language/Fixpoint/Smt/Theories.hs

```diff
@@ -590,7 +590,7 @@ interpSym :: Symbol -> Raw -> Sort -> (Symbol, TheorySymbol)
 interpSym x n t = (x, Thy x n t Theory)
 
 maxLamArg :: Int
-maxLamArg = 7
+maxLamArg = 20
```

Comment:
This `maxLamArg` looks rather pointless, it is not used for anything other than exporting it. If you know what is needed for, it would be helpful to document it here, as well as what is the consequence of using lambdas that exceed this amount of arguments.

Reply:
I added a comment to explain how it's used or at least what I understood about it, I don't know if it's used somewhere else